cd_scripts/install_test.sh

```diff
@@ -0,0 +1,107 @@
+    echo 'deb http://packages.cloud.google.com/apt apt-transport-artifact-registry-stable main' | sudo tee -a /etc/apt/sources.list.d/artifact-registry.list
+    sudo apt update
+    sudo apt install apt-transport-artifact-registry
+    echo 'deb ar+https://us-apt.pkg.dev/projects/gcs-fuse-prod $(lsb_release -cs) main' | sudo tee -a /etc/apt/sources.list.d/artifact-registry.list
+    sudo apt update
+
+    # Install released gcsfuse version
+    sudo apt install -y gcsfuse=$(sed -n 1p details.txt) -t gcsfuse-$(lsb_release -cs) |& tee -a ~/logs.txt
+else
+#  For rhel and centos
+    sudo yum makecache
+    sudo yum -y install yum-plugin-artifact-registry
+sudo tee -a /etc/yum.repos.d/artifact-registry.repo << EOF
+[gcsfuse-el7-x86-64]
+name=gcsfuse-el7-x86-64
+baseurl=https://asia-yum.pkg.dev/projects/gcs-fuse-prod/gcsfuse-el7-x86-64
+enabled=1
+repo_gpgcheck=0
+gpgcheck=1
+gpgkey=https://packages.cloud.google.com/yum/doc/yum-key.gpg
+    https://packages.cloud.google.com/yum/doc/rpm-package-key.gpg
+EOF
+    sudo yum makecache
+    sudo yum -y --enablerepo=gcsfuse-el7-x86-64 install gcsfuse-$(sed -n 1p details.txt)-1 |& tee -a ~/logs.txt
+fi
+
+# Verify gcsfuse version (successful installation)
+gcsfuse --version |& tee version.txt
+installed_version=$(echo $(sed -n 1p version.txt) | cut -d' ' -f3)
+if grep -q $installed_version details.txt; then
+    echo "GCSFuse latest version installed correctly." &>> ~/logs.txt
+else
+    echo "Failure detected in latest gcsfuse version installation." &>> ~/logs.txt
+fi
+
+# Uninstall gcsfuse latest version and install old version
+if grep -q ubuntu details.txt || grep -q debian details.txt;
+then
+    sudo apt remove -y gcsfuse
+    sudo apt install -y gcsfuse=0.42.5 -t gcsfuse-$(lsb_release -cs) |& tee -a ~/logs.txt
+else
+    sudo yum -y remove gcsfuse
+    sudo yum -y install gcsfuse-0.42.5-1 |& tee -a ~/logs.txt
+fi
+
+# verify old version installation
+gcsfuse --version |& tee version.txt
+installed_version=$(echo $(sed -n 1p version.txt) | cut -d' ' -f3)
+if [ $installed_version == "0.42.5" ]; then
+    echo "GCSFuse old version (0.42.5) installed successfully" &>> ~/logs.txt
+else
+    echo "Failure detected in GCSFuse old version installation." &>> ~/logs.txt
+fi
+
+# Upgrade gcsfuse to latest version
+if grep -q ubuntu details.txt || grep -q debian details.txt;
+then
+    sudo apt install --only-upgrade gcsfuse |& tee -a ~/logs.txt
+else
+    sudo yum -y upgrade gcsfuse |& tee -a ~/logs.txt
+fi
+
+gcsfuse --version |& tee version.txt
+installed_version=$(echo $(sed -n 1p version.txt) | cut -d' ' -f3)
+if grep -q $installed_version details.txt; then
+    echo "GCSFuse successfully upgraded to latest version $installed_version." &>> ~/logs.txt
+else
+    echo "Failure detected in upgrading to latest gcsfuse version." &>> ~/logs.txt
+fi
+
+if grep -q Failure ~/logs.txt; then
```

Comment:
I don't think it will happen here.
Incase gcsfuse isn't installed properly, we will get the following error on running `gcsfuse --version` command:
`bash: /usr/bin/gcsfuse: No such file or directory`
Since the version won't be present, the log file will have failure logs.

I think we might miss some logs here on setting -e